auth_session_timeout/models/res_users.py

```diff
@@ -17,28 +17,70 @@
 class ResUsers(models.Model):
     _inherit = 'res.users'
 
-    def _check_session_validity(self, db, uid, passwd):
+    def _auth_timeout_urls_get(self, session, db, uid):
+        # Pluggable method for calculating ignored urls
+        # Defaults to stored config param
+        param_obj = self.pool['ir.config_parameter']
+        return param_obj._auth_timeout_get_parameter_urls(db)
+
+    def _auth_timeout_deadline_calculate(self, session, db, uid):
+        # Pluggable method for calculating timeout deadline
+        # Defaults to current time minus delay using delay stored as session param
+        param_obj = self.pool['ir.config_parameter']
+        delay = param_obj._auth_timeout_get_parameter_delay(db)
+        if delay == False or delay <= 0:
+            return False
+        return time() - delay
+
+    def _auth_timeout_session_validity_check(self, db, uid, passwd):
         if not request:
             return
+
         session = request.session
-        session_store = root.session_store
-        param_obj = self.pool['ir.config_parameter']
-        delay, urls = param_obj.get_session_parameters(db)
-        deadline = time() - delay
-        path = session_store.get_session_filename(session.sid)
-        try:
-            if getmtime(path) < deadline:
-                if session.db and session.uid:
-                    session.logout(keep_db=True)
-            elif http.request.httprequest.path not in urls:
-                # the session is not expired, update the last modification
-                # and access time.
+
+        # Calculate Deadline DateTime
+        deadline = self._auth_timeout_deadline_calculate(session, db, uid)
+
+        # Check if deadline expired
+        expired = False
+        if deadline != False:
+            path = root.session_store.get_session_filename(session.sid)
+            try:
+                expired = getmtime(path) < deadline
+            except OSError:
+                pass
+
+        # Try to expire the session
+        if expired:
+            expired = self._auth_timeout_session_expire(session, db, uid)
+
+        # If not aborted, all done.
+        if expired:
+            return
+        
+        # Otherwise, fetch urls to ignore
+        urls = self._auth_timeout_urls_get(session, db, uid)
+
+        # Update session last modified and access times
+        if http.request.httprequest.path not in urls:
+            try:
                 utime(path, None)
-        except OSError:
-            pass
+            except OSError:
+                pass
```

Comment:
I know it's not your code, but you're already here. Never pass silently - `_logger.debug`